File: Competitions/JAN17/CAPIMOVE.cpp
#include<stdio.h>

struct Planet{
	int P;
	int connected[101];
	int numberofconn;
	
};

int main(){
	int i,j,k,T,N,Pi,u,v;
	scanf("%d",&T);
	while(T--){
		scanf("%d",&N);
		Planet X[N];
		int infected[N]={0};
		for(i=0;i<N;i++)scanf("%d",&X.P);
		for(i=0;i<N-1;i++){
		
			scanf("%d %d",&u,&v);
			X[u-1].connected[X[u-1].numberofconn]=v-1;
			X[u-1].numberofconn++;
			X[v-1].connected[X[v-1].numberofconn]=u-1;
			X[v-1].numberofconn++;
		}
		
		for(i=0;i<N;i++){
			for(j=0;j<N;j++)infected[j]=0;
			infected[i]=1;
			for(j=0;j<X[i].numberofconn;j++){
				infected[X[i].connected[j]]=1;
			}
			int maxpop=0;
			int index=0;
			for(j=0;j<N;j++){
				if(infected[j]==0){
					if(maxpop<X[j].P){
						maxpop=X[j].P;
						index=j;
					}
				}
			}
		printf("%d ",index+1);
		}
	}
}

File: Competitions/JAN17/CAPIMOVE.py
def infectedplanet(i,n,P,connections,neighbours):
    infected=[0]*n
    infected[i]=1
    for jk in range(neighbours[i]):
        infected[connections[i][jk]]=1
    maxpop=0
    index=0
    for k in range(n):
        if not infected[k]:
            if P[k]>maxpop:
                maxpop=P[k]
                index=k
                print('index updated',index)
    return [index+1,infected]

T=int(input())
for t in range(T):
    n=int(input())
    P=list(map(int,input().split(" "))) #populations 
    connections=[[0]*n]*n #neighbour indexes of planet i
    neighbours=[0]*n #number of neighbours of planet i
    #infected=[0]*n
    '''for _ in range(n-1):
        a,b=map(int,input().split(" "))
        connections[a-1][neighbours[a-1]]=(b-1)
        neighbours[a-1]+=1
        connections[b-1][neighbours[b-1]]=(a-1)
        neighbours[b-1]+=1
        print(connections,"\n\n",neighbours)'''
    for _ in range(n-1):
        a,b=map(int,input().split(" "))
        connections[a-1].append(b-1)
        neighbours[a-1]+=1
        connections[b-1].append(a-1)
        neighbours[b-1]+=1
        print(connections,'\n',neighbours)
    for it in range(n):
        print(infectedplanet(it,n,P,connections,neighbours),end=' ')
#planet 4 5 6
        
